### PR TITLE
feat: new conversation creation system messages (AR-910)

### PR DESCRIPTION
**Pull request by yamilmedina**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-910" title="AR-910" target="_blank">StoryAR-910</a>  Initial Group Introduction System Message
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Currently, we don't show a system message when starting a group conversation.

### Solutions

This PR:
- Add the system message info, in conversation creation flow, about members the conversation was started with.
- Adds initial system message about conversation creation date
- Prepares the code for later offline backend handling for conversation creation with remote participants.

Needs to be merged **after** merging:
- https://github.com/wireapp/kalium/pull/1762

### Attachments

https://github.com/wireapp/wire-android-reloaded/assets/5806454/7ec16409-6b5e-4c23-be07-bbc04cfc34e3

#### Test Coverage (Optional)

- [x] I have added automated test to this contribution

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1828/2/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.2.0(6649681)(6649681)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.2.0(6649681)(6649681)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/mapper/SystemMessageContentMapper.kt [hunk 1]
```diff
@@ -54,6 +56,13 @@ class SystemMessageContentMapper @Inject constructor(
         is MessageContent.ConversationReceiptModeChanged -> mapConversationReceiptModeChanged(message.senderUserId, content, members)
         is MessageContent.HistoryLost -> mapConversationHistoryLost()
         is MessageContent.ConversationMessageTimerChanged -> mapConversationTimerChanged(message.senderUserId, content, members)
+        is MessageContent.ConversationCreated -> mapConversationCreated(message.date, members)
+    }
+
+    private fun mapConversationCreated(date: String, userList: List<User>): UIMessageContent.SystemMessage {
+        return UIMessageContent.SystemMessage.ConversationMessageCreated(
+            date.formatMediumDateShortTime().orDefault(date).toUpperCase()
+        )
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/mapper/SystemMessageContentMapper.kt]: are we using `userList` here? 🤔

**Review by alexandreferris — COMMENTED** [review]
Looking great!

I just have one tiny question 😅

File: app/src/main/kotlin/com/wire/android/mapper/SystemMessageContentMapper.kt (hunk 1 above)
Comment (yamilmedina) [reply app/src/main/kotlin/com/wire/android/mapper/SystemMessageContentMapper.kt]: Good catch, fixed, now I'm using it, for preparation for later use on conversation creation event.

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1828/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.2.0(6655956)(6655956)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.2.0(6655956)(6655956)-staging-compat.apk)

**Review by alexandreferris — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-1828/6/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.2.0(6657496)(6657496)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.2.0(6657496)(6657496)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/util/DateTimeUtil.kt [hunk 2]
```diff
@@ -48,13 +48,25 @@ private val readReceiptDateTimeFormat = SimpleDateFormat(
     Locale.getDefault()
 ).apply { timeZone = TimeZone.getDefault() }
 
+private val dateTimeFormatWithDayOfWeek = SimpleDateFormat(
+    "EEE, MMM dd, hh:mm a",
```
Comment (vitorhugods) [inline app/src/main/kotlin/com/wire/android/util/DateTimeUtil.kt]: I just wonder how that would look in different languages.

`Day of the week, month, day of the month`

At least in Portuguese that's weird. Also AM/PM is a bit weird. And wouldn't follow user settings for this.

We could use [DateFormat](https://developer.android.com/reference/java/text/DateFormat.html):

```kotlin
DateFormat.getDateTimeInstance(DateFormat.FULL, DateFormat.SHORT).format(date)
```

`DateFormat.FULL` for Dates will include the week day.
`DateFormat.SHORT` for Time will keep only hours and minutes.


But it should respect Locale without we needing to create different patterns for different languages.

**Review by vitorhugods — CHANGES_REQUESTED** [review]
Looks awesome!

Just one suggestion for date/time formatting.

**Review by MohamadJaara — APPROVED** [review]

File: app/src/main/kotlin/com/wire/android/util/DateTimeUtil.kt (hunk 2 above)
Comment (yamilmedina) [reply app/src/main/kotlin/com/wire/android/util/DateTimeUtil.kt]: Great idea and makes sense to me, more easy to use the standard one. Changed

**Comment by AndroidBob** [comment]
Build [7](https://10.10.124.134/job/wire-android-reloaded/job/PR-1828/7/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.2.0(6657911)(6657911)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.2.0(6657911)(6657911)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [8](https://10.10.124.134/job/wire-android-reloaded/job/PR-1828/8/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.2.0(6658531)(6658531)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.2.0(6658531)(6658531)-staging-compat.apk)

File: buildSrc/src/main/kotlin/scripts/quality.gradle.kts [hunk 3]
```diff
@@ -64,7 +64,7 @@ val detektAll by tasks.registering(Detekt::class) {
     config.setFrom("$rootDir/config/detekt/detekt.yml")
 
     include("**/*.kt")
-    exclude("**/*.kts", "*/build/*", "/buildSrc")
+    exclude("**/*.kts", "*/build/*", "/buildSrc", "**/test/**/*")
```
Comment (vitorhugods) [inline buildSrc/src/main/kotlin/scripts/quality.gradle.kts]: Is this necessary?

**Review by vitorhugods — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [10](https://10.10.124.134/job/wire-android-reloaded/job/PR-1828/10/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.2.0(6659717)(6659717)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.2.0(6659717)(6659717)-staging-compat.apk)

**Pull request merged**